src/blockcipher/aes/scl_aes_ccm.c

```diff
@@ -67,36 +67,64 @@ int32_t scl_aes_ccm_init(const metal_scl_t *const scl_ctx, aes_auth_ctx_t *const
     }
 
     // get ccmt value
+    if (tag_byte_len > BLOCK128_NB_BYTE)
+    {
+        return (SCL_INVALID_INPUT);
+    }
     ccmt = ccmt_tab[tag_byte_len];
     if ((uint8_t)-1 == ccmt) 
     {
         return (SCL_INVALID_INPUT);
     }
 
     // determine ccmq value
-    // 2⁸
-    if (pld_byte_len < 0x100) 
+    // 2^16
+    if ( pld_byte_len < (((uint64_t)1 << (2*8))) ) 
     {
-        ccmq = 1;
+        ccmq = 2;
     } 
     else
     {
-        // 2¹⁶
-        if (pld_byte_len < 0x10000) 
+        // 2^24
+        if ( pld_byte_len < (((uint64_t)1 << (3*8))) )
         {
-            ccmq = 2;
+            ccmq = 3;
         } 
         else
+        // 2^32
         {
-            // 2²⁴
-            if (pld_byte_len < 0x1000000) 
+            if ( pld_byte_len < (((uint64_t)1 << (4*8))) )
             {
-                ccmq = 3;
+                ccmq = 4;
             } 
             else
-            // 2³²
+            // 2^40
             {
-                ccmq = 4;
+                if ( pld_byte_len < (((uint64_t)1 << (5*8))) )
+                {
+                    ccmq = 5;
+                } 
+                else
+                // 2^48
+                {
+                    if ( pld_byte_len < (((uint64_t)1 << (6*8))) )
+                    {
+                        ccmq = 6;
+                    } 
+                    else
+                    // 2^56
+                    {
+                        if ( pld_byte_len < (((uint64_t)1 << (7*8))) )
+                        {
+                            ccmq = 7;
+                        } 
+                        else
+                        // 2^64
+                        {
+                            ccmq = 8;
+                        }

```

Comment:
Can we use if else statement instead of if imbrication.

Reply:
Changed